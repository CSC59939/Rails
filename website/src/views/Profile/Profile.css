.profile-page {
  width: 100%;
  height: 100%;
  padding: 20px;
  display: flex;
  flex-direction: row;
}

.profile-card {
  flex: 1;
  height: fit-content !important;
  margin: 0 10px !important;
  border-radius: 15px !important;
  filter: drop-shadow(0 1px 2px rgba(190, 190, 190, 0.5));
}

.profile-card .cust-button,
.cust-input {
  margin-top: 10px !important;
}

.profile-card .cust-input {
  border-radius: 15px !important;
}

.profile-card .cust-button {
  border-radius: 15px !important;
}

.center-loading-icon {
  display: flex;
  height: 100%;
  width: 100%;
  justify-content: center;
  align-items: center;
}

.academics-subtitle {
  margin: 0;
  font-size: 18px;
  margin-top: 20px;
  font-weight: 500;
  margin-bottom: 7.5px;
}

.ant-modal-body {
  max-height: 50vh !important;
  overflow-y: auto !important;
}
