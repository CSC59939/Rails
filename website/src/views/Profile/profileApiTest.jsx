const fake_data =     {
        "message" : "User profile found",
        "userData" : {
            "displayName": "Full Name",
            "email": "fullname@email.com",
            "type": "student",
            "universities": {
                "CUNY City College" : {
                    "classUid1" : {
                        "name": "CSC 59939 (L)",
                        "description" : "Topics In Software Engineering",
                        "instructorName" : "Teacher Name",
                        "meetingTimes" : {
                            "from": "18:30",
                            "to": "21:00"
                        },
                        "meetingDays" : {
                            "Monday": false,
                            "Tuesday": false,
                            "Wednesday": true,
                            "Thursday": false,
                            "Friday": false,
                            "Saturday": false,
                            "Sunday": false,
                        },
                        "events" : {
                            "eventUid" : {
                                "postedDate" : "2018-11-06T23:37:04.888Z",
                                "dueDate" : "2018-12-06T23:37:04.888Z",
                                "title" : "First Release",
                                "description" : "asdfkaljsdfas",
                                "priority" : 2,
                            }
                        }
                    },
                    "classUid2" : {
                        "name": "CSC 59999 (L)",
                        "description" : "Topics In Software Engineering",
                        "instructorName" : "Teacher Name",
                        "meetingTimes" : {
                            "from": "18:30",
                            "to": "21:00"
                        },
                        "meetingDays" : {
                            "Monday": false,
                            "Tuesday": false,
                            "Wednesday": true,
                            "Thursday": false,
                            "Friday": false,
                            "Saturday": false,
                            "Sunday": false,
                        },
                        "events" : {
                            "eventUid" : {
                                "postedDate" : "2018-11-06T23:37:04.888Z",
                                "dueDate" : "2018-12-06T23:37:04.888Z",
                                "title" : "First Release",
                                "description" : "asdfkaljsdfas",
                                "priority" : 2,
                            }
                        }
                    }
                },
                "CUNY Baruch College" : {
                    "classUid1" : {
                        "name": "CSC 59939 (L)",
                        "description" : "Topics In Software Engineering",
                        "instructorName" : "Teacher Name",
                        "meetingTimes" : {
                            "from": "18:30",
                            "to": "21:00"
                        },
                        "meetingDays" : {
                            "Monday": false,
                            "Tuesday": false,
                            "Wednesday": true,
                            "Thursday": false,
                            "Friday": false,
                            "Saturday": false,
                            "Sunday": false,
                        },
                        "events" : {
                            "eventUid" : {
                                "postedDate" : "2018-11-06T23:37:04.888Z",
                                "dueDate" : "2018-12-06T23:37:04.888Z",
                                "title" : "First Release",
                                "description" : "asdfkaljsdfas",
                                "priority" : 2,
                            }
                        }
                    },
                    "classUid2" : {
                        "name": "CSC 59999 (L)",
                        "description" : "Topics In Software Engineering",
                        "instructorName" : "Teacher Name",
                        "meetingTimes" : {
                            "from": "18:30",
                            "to": "21:00"
                        },
                        "meetingDays" : {
                            "Monday": false,
                            "Tuesday": false,
                            "Wednesday": true,
                            "Thursday": false,
                            "Friday": false,
                            "Saturday": false,
                            "Sunday": false,
                        },
                        "events" : {
                            "eventUid" : {
                                "postedDate" : "2018-11-06T23:37:04.888Z",
                                "dueDate" : "2018-12-06T23:37:04.888Z",
                                "title" : "First Release",
                                "description" : "asdfkaljsdfas",
                                "priority" : 2,
                            }
                        }
                    }
                }
            }
        }
    }
    window.fetch('https://us-central1-rails-students.cloudfunctions.net/getprofile', {
          method: 'POST',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(reqData),
        })
    .then((res) => {
        if (res.status === 200) {
           // success case
           console.log(res)
        } else {
            // check for other codes if needed,
            // can otherwise just show the message
            message.info(res.message); // import { message } from 'antd';
        }
    })
    .catch((err) => {
        // Error sending request. Client side should handle
    });

const getName = data => data.userData.displayName;
const getEmail = data => data.userData.email;
const getUniversities = (data) => {
  let result = [];
  for (let university in data.userData.universities) {
    //console.log(university); // loop each university
    let name = university;
    result.push({name:university,course:data.userData.universities[university]});
  }
  return result;
}
console.log(getName(fake_data))
console.log(getEmail(fake_data))
console.log(getUniversities(fake_data));
