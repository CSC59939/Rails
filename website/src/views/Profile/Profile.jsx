import React, { PureComponent } from 'react';
import {
  Button, Icon, Input, Form, message, Card, Collapse,
} from 'antd';
import './Profile.css';
import firebase from 'firebase/app';
import 'firebase/auth';
import { WithProtectedView } from '../../hoc';

class Profile extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      gotUserProfile: false,
      userData: null,
    };
    this.changePassword = this.changePassword.bind(this);
  }

  componentDidMount() {
    firebase.auth().currentUser
      .getIdToken(true)
      .then((idToken) => {
        const { uid } = firebase.auth().currentUser;
        const reqData = { uid, idToken };
        const profileData = fetch('https://us-central1-rails-students.cloudfunctions.net/getprofile', {
          method: 'POST',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(reqData),
        }).then((result) => {
          if (result.status === 200) {
            return result.json();
          }
          message.error(result.message);
        }).catch((err) => {
          console.log(err);
        });
        profileData.then((data) => {
          message.success(data.message);
          console.log(data.userData);
          this.setState({ userData: data.userData, gotUserProfile: true });
        });
      });
  }

  changePassword() {
    const { userData } = this.state;
    const oldPassword = document.getElementById('oldPasswordInput').value;
    const newPassword = document.getElementById('newPasswordInput').value;
    firebase.auth().signInWithEmailAndPassword(userData.email, oldPassword)
      .then((user) => {
        if (user) {
          firebase.auth().currentUser.updatePassword(newPassword).then(() => {
            message.success('Changed Password, please sign in again.', 1.5, () => { window.location = '/signout'; });
          }).catch((err) => {
            console.log(err);
          });
        }
      });
  }

  render() {
    const { userData, gotUserProfile } = this.state;
    return (
      <div className="profile-page">
        {
        gotUserProfile
          ? (
            <div className="profile-page">

              <Card
                className="profile-card"
                title="About You"
              >
                <Input className="cust-input" placeholder="Full Name" value={userData ? userData.displayName : ''} />
                <Input className="cust-input" placeholder="E-Mail" value={userData ? userData.email : ''} />
                <Input className="cust-input" placeholder="Old Password" id="oldPasswordInput" />
                <Input className="cust-input" placeholder="New Password" id="newPasswordInput" />
                <Button className="cust-button" type="primary" onClick={this.changePassword} block>Change Password</Button>
              </Card>
              <Card
                className="profile-card"
                title="Your Academics"
              >
                {
            userData.type === 'student'
              ? <Button style={{ marginBottom: 10 }} href="/join/class" type="primary" block>Join Class</Button>
              : <Button style={{ marginBottom: 10 }} href="/create/class" type="primary" block>Create Class</Button>
          }
                {
                Object.keys(userData.universities).length === 0
                  ? (
                    <p>You haven't joined any universities/classes yet.</p>
                  )
                  : (
                    <Collapse>
                      {
                    Object.keys(userData.universities).map(university => (
                      <Collapse.Panel header={university}>
                        {
                            Object.keys(userData.universities[university]).map(classUid => (
                              <Card
                                title={userData.universities[university][classUid].name}
                              >
                                <p>{userData.universities[university][classUid].description}</p>
                              </Card>
                            ))
                          }
                      </Collapse.Panel>
                    ))
                  }
                    </Collapse>
                  )
            }
              </Card>
            </div>
          )
          : (
            <div style={{
              width: '100%', height: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center',
            }}
            >
              <Icon className="protected-view-loading" type="loading" theme="outlined" />
            </div>
          )
      }
      </div>
    );
  }
}

const ProtectedProfile = WithProtectedView(Profile);
export { Profile, ProtectedProfile };
