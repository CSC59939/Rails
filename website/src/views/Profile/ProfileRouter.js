import React from 'react';
import {
  BrowserRouter as Router, Route,
} from 'react-router-dom';
import { Dashboard } from '../../components';
import Profile from './Profile';
import { WithProtectedView } from '../../hoc';

const ProfileRouter = () => (
  <Router>
    <Dashboard>
      <Route exact path="/profile" component={Profile} />
    </Dashboard>
  </Router>
);

const ProtectedProfileRouter = WithProtectedView(ProfileRouter);
export { ProtectedProfileRouter, ProfileRouter };
