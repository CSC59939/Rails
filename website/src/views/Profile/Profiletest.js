import React from 'react';
import renderer from 'react-test-renderer';
import { Profile } from './Profile';
import '../../utils/tests/test.css';
import firebase from 'firebase/app';
import 'firebase/auth';

describe('Profile View', () => {
  it('Expect the Profile page to match snap shot', () => { // eslint-disable-line no-undef
    const component = renderer.create(
      <Profile />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot(); // eslint-disable-line no-undef
  });
});
