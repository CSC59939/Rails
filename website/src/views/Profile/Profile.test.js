import React from 'react';
import renderer from 'react-test-renderer';
import { Profile } from './Profile';
import '../../utils/tests/test.css';

describe('Profile View', () => {
  it('Expect the Profile page to match snap shot', () => { // eslint-disable-line no-undef
    const component = renderer.create(
      Profile,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot(); // eslint-disable-line no-undef
  });
});
