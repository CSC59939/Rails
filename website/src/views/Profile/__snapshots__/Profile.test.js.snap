// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile View Expect the Profile page to match snap shot 1`] = `
<div
  className="Profile"
>
  <div
    className="Body"
  >
    <div>
      <p
        className="AboutYou"
      >
        About You
      </p>
      <p
        className="YourAcademic"
      >
        Your Academic
      </p>
    </div>
    <form
      className="ant-form ant-form-horizontal Body_AboutYou"
      onSubmit={[Function]}
    >
      <input
        className="ant-input AboutYou_input input_UserName"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="User Name"
        style={
          Object {
            "width": "70%",
          }
        }
        type="text"
      />
      <input
        className="ant-input AboutYou_input input_UserEmail"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="User Email(Not EDITABLE)"
        style={
          Object {
            "width": "70%",
          }
        }
        type="email"
      />
      <input
        className="ant-input AboutYou_input input_OldPassword"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Old Password(EMPTY INITIALLY)"
        style={
          Object {
            "width": "70%",
          }
        }
        type="password"
      />
      <input
        className="ant-input AboutYou_input input_NewPassword"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="New Password(EMPTY INITIALLY)"
        style={
          Object {
            "width": "70%",
          }
        }
        type="password"
      />
      <button
        className="ant-btn Button_ChangePassword"
        onClick={[Function]}
        style={
          Object {
            "color": "#87ceeb",
          }
        }
        type="button"
      >
        <span>
          Change Password
        </span>
      </button>
    </form>
    <div
      className="Body_YourAcademics"
    >
      <div
        className="Body_University"
      >
        <h3
          className="UniversityName"
        >
          University 1
        </h3>
        <ul>
          <li>
            ClassA 
            <i
              className="anticon anticon-select IconSelect"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="select"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h360c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H184V184h656v320c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V144c0-17.7-14.3-32-32-32zM653.3 599.4l52.2-52.2a8.01 8.01 0 0 0-4.7-13.6l-179.4-21c-5.1-.6-9.5 3.7-8.9 8.9l21 179.4c.8 6.6 8.9 9.4 13.6 4.7l52.4-52.4 256.2 256.2c3.1 3.1 8.2 3.1 11.3 0l42.4-42.4c3.1-3.1 3.1-8.2 0-11.3L653.3 599.4z"
                />
              </svg>
            </i>
          </li>
          <li>
            ClassB 
            <i
              className="anticon anticon-select IconSelect"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="select"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h360c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H184V184h656v320c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V144c0-17.7-14.3-32-32-32zM653.3 599.4l52.2-52.2a8.01 8.01 0 0 0-4.7-13.6l-179.4-21c-5.1-.6-9.5 3.7-8.9 8.9l21 179.4c.8 6.6 8.9 9.4 13.6 4.7l52.4-52.4 256.2 256.2c3.1 3.1 8.2 3.1 11.3 0l42.4-42.4c3.1-3.1 3.1-8.2 0-11.3L653.3 599.4z"
                />
              </svg>
            </i>
          </li>
        </ul>
      </div>
      <div
        className="Body_University"
      />
      <button
        className="ant-btn Button_JoinAnotherClass"
        onClick={[Function]}
        style={
          Object {
            "color": "#87ceeb",
          }
        }
        type="button"
      >
        <span>
          Join Another Class
        </span>
      </button>
      <button
        className="ant-btn Button_JoinAnotherUniversity ant-btn-primary"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
          }
        }
        type="button"
      >
        <span>
          Join Another University 
        </span>
      </button>
    </div>
  </div>
</div>
`;
